stablehlo/tests/interpret_sign.mlir

```diff
@@ -0,0 +1,168 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_si4
+func.func @sign_op_test_si4() -> tensor<3xi4> {
+  %operand = stablehlo.constant dense<[-7, 0, 7]> : tensor<3xi4>
+  %result = stablehlo.sign %operand : tensor<3xi4>
+  func.return %result : tensor<3xi4>
+  // CHECK-NEXT: tensor<3xi4>
+  // CHECK-NEXT: -1 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_si8
+func.func @sign_op_test_si8() -> tensor<3xi8> {
+  %operand = stablehlo.constant dense<[-127, 0, 127]> : tensor<3xi8>
+  %result = stablehlo.sign %operand : tensor<3xi8>
+  func.return %result : tensor<3xi8>
+  // CHECK-NEXT: tensor<3xi8>
+  // CHECK-NEXT: -1 : i8
+  // CHECK-NEXT: 0 : i8
+  // CHECK-NEXT: 1 : i8
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_si16
+func.func @sign_op_test_si16() -> tensor<3xi16> {
+  %operand = stablehlo.constant dense<[-32767, 0, 32767]> : tensor<3xi16>
+  %result = stablehlo.sign %operand : tensor<3xi16>
+  func.return %result : tensor<3xi16>
+  // CHECK-NEXT: tensor<3xi16>
+  // CHECK-NEXT: -1 : i16
+  // CHECK-NEXT: 0 : i16
+  // CHECK-NEXT: 1 : i16
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_si32
+func.func @sign_op_test_si32() -> tensor<3xi32> {
+  %operand = stablehlo.constant dense<[-2147483647, 0, 2147483647]> : tensor<3xi32>
+  %result = stablehlo.sign %operand : tensor<3xi32>
+  func.return %result : tensor<3xi32>
+  // CHECK-NEXT: tensor<3xi32>
+  // CHECK-NEXT: -1 : i32
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 1 : i32
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_si64
+func.func @sign_op_test_si64() -> tensor<3xi64> {
+  %operand = stablehlo.constant dense<[-9223372036854775807, 0, 9223372036854775807]> : tensor<3xi64>
```

Comment:
I think you meant the negative number to be `-9223372036854775808`. In general, let's use more readable numbers as whether we test maximal/minimal representable values does not give us any additional information and avoids typo. Proposed numbers:
```
%operand = stablehlo.constant dense<[-1, 0, 1]> : tensor<3xi64>
```